src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/SelectInstanceTypeSection.tsx

```diff
@@ -1,46 +1,63 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useMemo, useState } from 'react';
 
 import { useInstanceTypeVMStore } from '@catalog/CreateFromInstanceTypes/state/useInstanceTypeVMStore';
-import { instanceTypeActionType } from '@catalog/CreateFromInstanceTypes/state/utils/types';
-import { Grid, GridItem } from '@patternfly/react-core';
+import Loading from '@kubevirt-utils/components/Loading/Loading';
+import { Grid, GridItem, Tab, Tabs } from '@patternfly/react-core';
 
-import CategoryCard from './components/category-card/components/CategoryCard/CategoryCard';
-import CustomMenu from './components/category-card/components/InstanceTypesMenu/components/CustomMenu';
-import { InstanceTypeCategory, InstanceTypeSize } from './utils/types';
-import { categoryDetailsMap } from './utils/utils';
+import { getInstanceTypeMenuItems } from '../AddBootableVolumeModal/components/VolumeMetadata/components/InstanceTypeDrilldownSelect/utils/utils';
 
-import './SelectInstanceTypeSection.scss';
+import RedHatSeriesMenuCard from './components/RedHatSeriesMenuCard/RedHatSeriesMenuCard';
+import UsersInstanceTypesList from './components/UsersInstanceTypeList/UsersInstanceTypeList';
+import useInstanceTypeCardMenuSection from './hooks/useInstanceTypeCardMenuSection';
+import { TabKey } from './utils/constants';
 
 const SelectInstanceTypeSection: FC = () => {
-  const { instanceTypeVMState, setInstanceTypeVMState } = useInstanceTypeVMStore();
-  const { selectedInstanceType } = instanceTypeVMState;
-  const handleSelect = (category: InstanceTypeCategory, size: InstanceTypeSize) => {
-    setInstanceTypeVMState({
-      type: instanceTypeActionType.setSelectedInstanceType,
-      payload: {
-        category,
-        size,
-        name: `${categoryDetailsMap[category]?.prefix}.${size}`,
-      },
-    });
+  const [activeTabKey, setActiveTabKey] = useState<TabKey>(TabKey.RedHat);
+
+  const {
+    instanceTypesAndPreferencesData: { instanceTypes, loaded },
+    instanceTypeVMState: { selectedInstanceType },
+  } = useInstanceTypeVMStore();
+  const menuItems = useMemo(() => getInstanceTypeMenuItems(instanceTypes), [instanceTypes]);
+
+  const menuProps = useInstanceTypeCardMenuSection();
+  const gridItems = useMemo(
```

Comment:
im not sure useMemo here is necessary, please check

Reply:
you're right, removed